### PR TITLE
Refs [Task][30561]: Handle event for Home screen

### PR DESCRIPTION
**Pull request by NguyenTaTrung**
## Related Tickets
- [#Task_30561: Handle event for Home screen](https://edu-redmine.sun-asterisk.vn/issues/30561)

## WHAT
- Handle event for Home screen

## HOW
- Bind data: get data pour into recyclerview
- Item click to display a list of exercises for each muscle group

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/streetworkout/ui/binding/ImageViewBinding.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import androidx.databinding.BindingAdapter
+import com.bumptech.glide.Glide
+import com.example.streetworkout.R
+import com.example.streetworkout.util.LinkConst.URL_IMAGE
+import com.google.android.material.imageview.ShapeableImageView
+
+@BindingAdapter("imageUrl")
+fun loadImage(imageView: ShapeableImageView, url: String) {
+    Glide.with(imageView.context).load(URL_IMAGE + url).into(imageView)
+}
```
Comment (thonc-2135) [inline app/src/main/java/com/example/streetworkout/ui/binding/ImageViewBinding.kt]: nếu lỗi thì sao nhỉ, nên có thêm defualt vào nhé

File: app/src/main/java/com/example/streetworkout/base/RxViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import androidx.lifecycle.ViewModel
+import io.reactivex.rxjava3.disposables.CompositeDisposable
+import io.reactivex.rxjava3.disposables.Disposable
+
+open class RxViewModel : ViewModel() {
+    private val disposables = CompositeDisposable()
+
+    fun addToDisposable(disposable: Disposable) {
+        disposables.addAll(disposable)
+    }
```
Comment (datdt-1935) [inline app/src/main/java/com/example/streetworkout/base/RxViewModel.kt]: dùng `addTo` của RxKotlin là được rồi, không cần cái này

File: app/src/main/java/com/example/streetworkout/ui/home/HomeAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import com.example.streetworkout.R
+import com.example.streetworkout.base.BaseAdapter
+import com.example.streetworkout.data.model.GroupMuscle
+
+class HomeAdapter : BaseAdapter<GroupMuscle, HomeViewHolder>(GroupMuscle.diffUtil) {
+    var onItemClick: (GroupMuscle, Int) -> Unit = { _, _ -> }
```
Comment (datdt-1935) [inline app/src/main/java/com/example/streetworkout/ui/home/HomeAdapter.kt]: để lên constructor

File: app/src/main/java/com/example/streetworkout/util/Constants.kt [hunk 4]
```diff
@@ -2,8 +2,15 @@ package com.example.streetworkout.util
...
 
 object DatabaseConst {
     const val DB_NAME = "street_workout_database"
 }
+
+object BundleConst {
+    const val BUNDLE_TITLE_EXERCISE = "BUNDLE_TITLE_EXERCISE"
+    const val BUNDLE_IMAGE_EXERCISE = "BUNDLE_IMAGE_EXERCISE"
+    const val BUNDLE_ID_EXERCISE = "BUNDLE_ID_EXERCISE"
+}
```
Comment (huytq-2422) [inline app/src/main/java/com/example/streetworkout/util/Constants.kt]: Mấy cái bundle này dùng để làm gì nhỉ

Comment (NguyenTaTrung) [reply app/src/main/java/com/example/streetworkout/util/Constants.kt]: T truyền dữ liệu fragment ý mà

File: app/src/main/java/com/example/streetworkout/ui/home/HomeFragment.kt [hunk 5]
```diff
@@ -1,14 +1,72 @@
...
+    }
+
+    private fun openDetailExerciseFragment(item: GroupMuscle) {
+        findNavController().navigate(
+            R.id.actionToDetailExerciseFragment, bundleOf(
+                BUNDLE_TITLE_EXERCISE to item.title,
+                BUNDLE_IMAGE_EXERCISE to item.image,
+                BUNDLE_ID_EXERCISE to item.id
+            )
+        )
```
Comment (huytq-2422) [inline app/src/main/java/com/example/streetworkout/ui/home/HomeFragment.kt]: val bundle = DetailExerciseFragment.getBundle(...) 
Trong hàm getBundle dùng DetailExerciseFragmentArgs

File: app/src/main/java/com/example/streetworkout/ui/home/HomeFragment.kt [hunk 6]
```diff
@@ -1,14 +1,72 @@
...
+import com.example.streetworkout.util.BundleConst.BUNDLE_TITLE_EXERCISE
+import com.example.streetworkout.util.showToast
+import kotlinx.android.synthetic.main.fragment_home.*
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class HomeFragment : BaseFragment() {
 
+    private val challengeAdapter = HomeAdapter { item, _ ->
+        openDetailExerciseFragment(item)
+    }
```
Comment (huytq-2422) [inline app/src/main/java/com/example/streetworkout/ui/home/HomeFragment.kt]:
```suggestion
    private val challengeAdapter = HomeAdapter(::openDetailExerciseFragment)
```

File: app/src/main/java/com/example/streetworkout/ui/binding/ImageViewBinding.kt (hunk 1 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/example/streetworkout/ui/binding/ImageViewBinding.kt]: Ok a

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**